internal/util/sessionutil/session_util.go

```diff
@@ -469,15 +469,27 @@ func (s *Session) processKeepAliveResponse(ch <-chan *clientv3.LeaseKeepAliveRes
 					log.Info("start try to KeepAliveOnce", zap.String("serverName", s.ServerName))
 					s.retryKeepAlive.Store(true)
 					// have to KeepAliveOnce before KeepAlive because KeepAlive won't throw error even when lease OT
-					keepAliveOnceResp, err := s.etcdCli.KeepAliveOnce(s.ctx, *s.leaseID)
+					var keepAliveOnceResp *clientv3.LeaseKeepAliveResponse
+					err := retry.Do(s.ctx, func() error {
+						resp, err := s.etcdCli.KeepAliveOnce(s.ctx, *s.leaseID)
```

Comment:
it should use a timeout context. s.ctx is no timeout still components exit.

Reply:
agreed